### PR TITLE
Refactor replay state into object info

### LINKED ISSUE #1719: [openxr] Refactor replay state/process to match Vulkan
Refactor the OpenXR replay Update functionality to more closely match the behavior in the Vulkan replay path. Currently consumer state is tracked separate from handle info.  Create derived class state object to support single creation, lookup, and cleanup.

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Fixes #1719 

Replay state no longer embedded in replay object, but added to derived object info structures for Session and Swapchain.

Implementation left in situ with original naming to simplify review.

Additional commits will migrate code to standalone files, potentially renaming them.

### CODE REVIEW

File: framework/decode/openxr_replay_session_state.cpp [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    // View space at view relative location
+    XrReferenceSpaceCreateInfo view_relative_ci = { XR_TYPE_REFERENCE_SPACE_CREATE_INFO,
+                                                    nullptr,
+                                                    XR_REFERENCE_SPACE_TYPE_VIEW,
+                                                    { { location.qx, location.qy, location.qz, location.qw },
+                                                      { location.x, location.y, location.z } } };
+
+    XrSpace view_relative_space = XR_NULL_HANDLE;
+    // TODO: Cleanup this reference space object at DestroySession time
```
Comment (MarkY-LunarG) [inline framework/decode/openxr_replay_session_state.cpp:90]: Does this need to be an issue?

**Review by MarkY-LunarG — APPROVED** [review]
Just a few nits/questions.

File: framework/decode/openxr_replay_session_state.cpp (hunk 1 above)
Comment (jzulauf-lunarg) [reply framework/decode/openxr_replay_session_state.cpp:90]: Implicitly part of  #1718

File: framework/decode/openxr_replay_swapchain_state.h [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    XrResult InitVirtualSwapchain(PointerDecoder<uint32_t>*                                 imageCountOutput,
+                                  StructPointerDecoder<Decoded_XrSwapchainImageBaseHeader>* capture_images);
+    XrResult InitVirtualSwapchain(PointerDecoder<uint32_t>*                                imageCountOutput,
+                                  StructPointerDecoder<Decoded_XrSwapchainImageVulkanKHR>* vk_capture_images);
+    XrResult AcquireSwapchainImage(uint32_t capture_index, uint32_t replay_index);
+    XrResult ReleaseSwapchainImage(StructPointerDecoder<Decoded_XrSwapchainImageReleaseInfo>* releaseInfo);
+    void     WaitedWithoutTimeout();
+
+  protected:
+    XrSwapchain swapchain_;
```
[the remark this answers is not in the document]
Comment (jzulauf-lunarg) [reply framework/decode/openxr_replay_swapchain_state.h:59]: done.

File: framework/decode/openxr_replay_swapchain_state.h [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    static void MapVulkanSwapchainImageFlags(XrSwapchainUsageFlags xr_flags, VkImageCreateInfo& info);
+    XrResult    InitSwapchainData(const XrSwapchainCreateInfo& xr_info, VulkanSwapchainInfo& vk_info);
+    XrResult    AcquireSwapchainImage(uint32_t capture_index, uint32_t replay_index, VulkanSwapchainInfo& swap_info);
+    XrResult    ReleaseSwapchainImage(StructPointerDecoder<Decoded_XrSwapchainImageReleaseInfo>* releaseInfo,
+                                      VulkanSwapchainInfo&                                       swap_info);
+
+    XrSwapchainCreateInfo                  create_info;
```
[the remark this answers is not in the document]
Comment (jzulauf-lunarg) [reply framework/decode/openxr_replay_swapchain_state.h:66]: done

**Review by jzulauf-lunarg — COMMENTED** [review]
will push nits plus Linux build fix.

**Pull request merged**